pkg/util/listUtils.go

```diff
@@ -18,7 +18,8 @@ func GetDuplicates(list []string) []string {
 	return duplicates
 }
 
-func Map[T, V any](ts []T, fn func(T) V) []V {
+// MapWithFunction takes a slice of a type T and a converter function in order to return a slice of type V.
+func MapWithFunction[T, V any](ts []T, fn func(T) V) []V {
```

Comment:
What was your intention with this renaming? Map is the standard name for this function in nearly all languages. Is there sth in Go to consider here?

Reply:
I agree with Alex here. That this function takes another function as an argument is visible from the signature already. This function is called `map` in any language I know:
- [Java](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/stream/Stream.html#map(java.util.function.Function))
- [Elixir](https://hexdocs.pm/elixir/Enum.html#map/2)
- [Haskell](https://hackage.haskell.org/package/base-4.19.0.0/docs/Prelude.html#v:map)
- [Rust](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map)

Reply:
The function signature contains a lot of noise because of the generics and their slice types. At first I thought this was a language function but it was not. IMO the inline functions do not add readability as well so I had to read a blob of code. Also a doc comment was missing so I needed some time to understand what was going on.

That was the reason why I renamed the function. If you all think it is better off the way it is I don't mind to revert it. That being said I like express that the code should be swiftly graspable. Maybe we can find a better solution but currently I cannot offer one.

Reply:
Yeah, I can understand that. Go generics are a mess to read. But IMO a doc comment should suffice. We could add a usage example if we're feeling fancy. Something like:

```go
// Returns a slice where each element is the result of invoking fn on each corresponding element of the given slice.
//
// Example:
//
//  strings := []string{"a", "ab", "abc"}
//  lengths := Map(strings, func(t string) int {
//  	return len(t)
//  })
//  fmt.Println(lengths)
//
// This should print: [1 2 3]
```

Reply:
Will do! Thanks for the suggestion, yall